Frameworks/Foundation/NSMutableSet.mm

```diff
@@ -139,11 +139,21 @@ - (void)removeObject:(id)object {
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (void)filterUsingPredicate:(NSPredicate*)predicate {
-    UNIMPLEMENTED();
+    if (predicate == NULL) {
+        return;
+    }
+    @synchronized(self) {
+        NSArray* objects = [self allObjects];
```

Comment:
Is this API guaranteed to be thread-safe?
If not, there's no reason to `@synchronized` here. None of the other APIs do.